app/views/targets/new.html.haml

```diff
@@ -1,42 +1,115 @@
 #googleMap
 %script{:src => "https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"}
 :javascript
+
+  var map_topic_image = {
+    'Football' : "#{image_path('football.svg')}",
+    'Travel' : "#{image_path('travel.svg')}",
+    'Politics' : "#{image_path('politics.svg')}",
+    'Art' : "#{image_path('art.svg')}",
+    'Dating' : "#{image_path('dating.svg')}",
+    'Music' : "#{image_path('music.svg')}",
+    'Movies' : "#{image_path('movies.svg')}",
+    'Series' : "#{image_path('series.svg')}",
+    'Food' : "#{image_path('food.svg')}"
+  };
+
   function myMap() {
+    navigator.geolocation.getCurrentPosition(showPosition);
+    function showPosition(position) {
+      var user_position = new google.maps.LatLng(position.coords.latitude, position.coords.longitude);
+      map.panTo(user_position);
+      new_marker.setPosition(user_position);
+      new_radius.setCenter(user_position);
+    }
+
     var mapProp = {
         center: new google.maps.LatLng(-34.906268,-56.186120),
         zoom:15,
     };
 
     var map = new google.maps.Map(document.getElementById("googleMap"), mapProp);
 
-    var marker = new google.maps.Marker({
+    var new_marker = new google.maps.Marker({
       position: new google.maps.LatLng(-34.906268,-56.186120),
       map: map,
       icon: {
     		url: "#{image_path('marker.svg')}"
     	}
     });
 
+    var new_radius = new google.maps.Circle({
+      strokeColor: '#efc638',
+      strokeWeight: 2,
+      fillColor: 'white',
+      fillOpacity: 0.7,
+      map: map,
+      center: new google.maps.LatLng(-34.906268,-56.186120),
+      radius: 100
+    });
+
     map.addListener('click', function(e) {
       placeMarker(e.latLng, map);
     });
 
     function placeMarker(location) {
-      if (marker == null) {
-        marker = new google.maps.Marker({
+      if (new_marker == null) {
+        new_marker = new google.maps.Marker({
         position: location,
         map: map
         });
+        new_radius = new google.maps.Circle({
+          strokeColor: '#efc638',
+          strokeWeight: 2,
+          fillColor: 'white',
+          fillOpacity: 0.7,
+          map: map,
+          center: new google.maps.LatLng(-34.906268,-56.186120),
+          radius: 100
+        });
       } else {
-         marker.setPosition(location);
+         new_marker.setPosition(location);
+         new_radius.setCenter(location);
+         new_radius.setRadius(parseInt($('#radius_select').val()));
       }
       $('#latitude').val(location.lat);
       $('#longitude').val(location.lng);
     }
+
+    $( document ).ready(function() {
+      $.getJSON("#{targets_path(format: :json)}", function(json) {
+        $(json).each(function() {
+          marker = new google.maps.Marker({
+            position: new google.maps.LatLng(this.latitude, this.longitude),
+            map: map,
+            icon: {
```

Comment:
Wrong identation, in this case you can just do: `icon: { url: map_topic_image[this.topic] }`